lnbits/core/views/node_api.py

```diff
@@ -0,0 +1,189 @@
+from typing import List, Optional
+
+import httpx
+from fastapi import APIRouter, Body, Depends, HTTPException
+from pydantic import BaseModel
+from starlette.status import HTTP_503_SERVICE_UNAVAILABLE
+
+from lnbits.decorators import check_admin, parse_filters
+from lnbits.nodes import get_node_class
+from lnbits.settings import settings
+
+from ...db import Filters, Page
+from ...nodes.base import (
+    ChannelPoint,
+    Node,
+    NodeChannel,
+    NodeInfoResponse,
+    NodeInvoice,
+    NodeInvoiceFilters,
+    NodePayment,
+    NodePaymentsFilters,
+    NodePeerInfo,
+    PublicNodeInfo,
+)
+from .. import core_app
+
+
+def require_node():
+    NODE = get_node_class()
+    if not NODE:
+        raise HTTPException(
+            status_code=400,
+            detail="Active backend does not implement Node API",
+        )
+    if not settings.lnbits_node_ui:
+        raise HTTPException(
+            status_code=404,
+            detail="Not enabled",
+        )
+    return NODE
+
+
+def check_public():
+    if not (settings.lnbits_node_ui and settings.lnbits_public_node_ui):
+        raise HTTPException(
+            status_code=404,
+            detail="Not enabled",
+        )
+
+
+node_api = APIRouter(prefix="/node/api/v1", dependencies=[Depends(check_admin)])
+public_node_api = APIRouter(
+    prefix="/node/public/api/v1", dependencies=[Depends(check_public)]
+)
+
+
+@node_api.get(
+    "/ok",
+    description="Check if node api can be enabled",
+    status_code=200,
+    dependencies=[Depends(require_node)],
+)
+async def api_get_ok():
+    pass
+
+
+@public_node_api.get("/info", response_model=PublicNodeInfo)
+async def api_get_public_info(node: Node = Depends(require_node)) -> PublicNodeInfo:
+    return await node.get_public_info()
```

Comment:
- this goes directly to the LN node, can't it be abused (since it is public)?

Reply:
- same for all `node: Node = Depends(require_node)`

Reply:
Its probably a good idea to cache it